src/main/java/net/minecraftforge/event/ForgeEventFactory.java

```diff
@@ -856,14 +857,14 @@ public static void onPlayerPostTick(Player player)
         MinecraftForge.EVENT_BUS.post(new TickEvent.PlayerTickEvent(TickEvent.Phase.END, player));
     }
 
-    public static void onPreWorldTick(Level world)
+    public static void onPreWorldTick(Level world, BooleanSupplier haveTime)
     {
-        MinecraftForge.EVENT_BUS.post(new TickEvent.WorldTickEvent(LogicalSide.SERVER, TickEvent.Phase.START, world));
+        MinecraftForge.EVENT_BUS.post(new TickEvent.WorldTickEvent(LogicalSide.SERVER, TickEvent.Phase.START, world, haveTime));
     }
 
-    public static void onPostWorldTick(Level world)
+    public static void onPostWorldTick(Level world, BooleanSupplier haveTime)
     {
-        MinecraftForge.EVENT_BUS.post(new TickEvent.WorldTickEvent(LogicalSide.SERVER, TickEvent.Phase.END, world));
+        MinecraftForge.EVENT_BUS.post(new TickEvent.WorldTickEvent(LogicalSide.SERVER, TickEvent.Phase.END, world, haveTime));
     }
```

Comment:
As a note for other reviewers, this hardcoded passing in of `LogicalSide.SERVER` to the world tick event seems to have been done all the way back in 95f0447881997533e5d352d96704e222bd5c36be when the tick events were first refactored to use the event system. So while this _seems_ to imply that the tick event may be fired on the logical client, it has never been fired on the logical client.